### PR TITLE
Add `getLocale()` method

### PR DESCRIPTION
**Pull request by Tigrov**
It needs to get current locale when page is rendering

`<html lang="<?= Html::encode($view->getLocale()) ?>">`

Also this will allow fix issue yiisoft/app#290

| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | -

### CODE REVIEW

**Review by samdark — APPROVED** [review]

File: src/ViewTrait.php [hunk 1]
```diff
@@ -189,6 +189,16 @@ public function withLocale(string $locale): static
...
     }
 
+    /**
+     * Get the specified locale code.
+     *
+     * @return string The locale code.
+     */
+    public function getLocale(): string
+    {
+        return $this->localeState->getLocale();
```
Comment (samdark) [inline src/ViewTrait.php:199]: Where's `localeState`?

Comment (Tigrov) [reply src/ViewTrait.php:199]: Good question.

It is inside `View` and `WebView`
https://github.com/yiisoft/view/blob/9547650ab106e8357613b75e38ac22a989d9f85c/src/WebView.php#L39-L45

And `state` also
https://github.com/yiisoft/view/blob/9547650ab106e8357613b75e38ac22a989d9f85c/src/ViewTrait.php#L228-L231

This task for a new issue.

Comment (vjik) [reply src/ViewTrait.php:199]: > This task for a new issue.

What is task?

`ViewTrait` is internal class that used in `View` and `WebView` only.

**Review by vjik — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,6 +9,7 @@
 - Chg #232: Deprecate `ViewInterface::withDefaultExtension()` and `ViewInterface::getDefaultExtension()` in favor of 
   `withFallbackExtension()` and `getFallbackExtensions()` (@rustamwin)
 - Bug #232: Fix render templates that contain dots in their name (@rustamwin)
+- Enh #242: Add `View::getLocale()` and `WebView::getLocale()` methods (@Tigrov)
```
Comment (vjik) [inline CHANGELOG.md:12]:
```suggestion
- New #242: Add `View::getLocale()` and `WebView::getLocale()` methods (@Tigrov)
```

File: src/ViewTrait.php [hunk 3]
```diff
@@ -189,6 +189,16 @@ public function withLocale(string $locale): static
         return $new;
     }
 
+    /**
+     * Get the specified locale code.
+     *
+     * @return string The locale code.
+     */
+    public function getLocale(): string
```
Comment (vjik) [inline src/ViewTrait.php:197]: Need to add new method to `ViewInterface` also.

**Review by vjik — CHANGES_REQUESTED** [review]

File: src/ViewTrait.php (hunk 1 above)
Comment (Tigrov) [reply src/ViewTrait.php:199]: > > This task for a new issue.
> 
> What is task?
> 
> `ViewTrait` is internal trait that used in `View` and `WebView` only.

Move `localeState` and `state` inside `ViewTrait`

File: src/ViewTrait.php (hunk 3 above)
Comment (Tigrov) [reply src/ViewTrait.php:197]: After that the PR becomes major.

Comment (vjik) [reply src/ViewTrait.php:197]: Yes, we should merge this PR before next major version.

**Review by vjik — APPROVED** [review]

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
👍

**Pull request merged**